core/src/main/java/tech/pegasys/eth2signer/core/signing/FcSecpArtifactSigner.java

```diff
@@ -58,4 +67,28 @@ public SecpArtifactSignature sign(final Bytes message) {
 
     return new SecpArtifactSignature(fcSignature);
   }
+
+  @Override
+  public boolean verify(final Bytes message, final ArtifactSignature signature) {
+    checkArgument(signature instanceof SecpArtifactSignature);
+    final SecpArtifactSignature secpArtifactSignature = (SecpArtifactSignature) signature;
+    final Signature signatureData = secpArtifactSignature.getSignatureData();
+
+    final ECDSASignature initialSignature =
+        new ECDSASignature(signatureData.getR(), signatureData.getS());
+    final ECDSASignature canonicalSignature = initialSignature.toCanonicalised();
+
+    final int recId = signatureData.getV().intValue();
+    final byte[] digest = Blake2b.sum256(message).toArrayUnsafe();
+    final BigInteger signaturePublicKey =
+        Sign.recoverFromSignature(recId, canonicalSignature, digest);
+    if (signaturePublicKey == null) {
+      LOG.error("Invalid signature");
```

Comment:
nit: can be more detailed such as `Unable to recover public key from signature`.

Reply:
done